packages/dd-trace/src/plugins/util/web.js

```diff
@@ -64,8 +64,12 @@ const web = {
 
     analyticsSampler.sample(span, config.analytics, true)
 
-    wrapEnd(req)
-    wrapEvents(req)
+    if (!req._datadog.instrumented) {
+      wrapEnd(req)
+      wrapEvents(req)
+    }
+
+    req._datadog.instrumented = true
```

Comment:
Shouldn't this be inside the `if` block? I know it doesn't matter because it will always be set, but we only *need* to set it inside the block, right?

Reply:
Correct. That's a small change, I'll do it now.

Reply:
Done.